provider/image.go

```diff
@@ -673,24 +794,59 @@ func getRegistryAddrFromImage(imgName string) (string, error) {
 	return addr, nil
 }
 
-func processLogLine(msg string) (string, error) {
-	var info string
-	var jm jsonmessage.JSONMessage
-	err := json.Unmarshal([]byte(msg), &jm)
-	if err != nil {
-		return info, fmt.Errorf("encountered error unmarshalling: %v", err)
+func (p *dockerNativeProvider) processLog(ctx context.Context, urn resource.URN,
+	in io.Reader, onAuxMessage func(json.RawMessage) (bool, string, error),
+) error {
+	decoder := json.NewDecoder(in)
+
+	var currentOffset int64
+	for {
+		var jm jsonmessage.JSONMessage
+		err := decoder.Decode(&jm)
+		if err != nil {
+			if err == io.EOF {
+				break
+			}
+			return fmt.Errorf("error parsing Docker output: %v", err)
+		}
+		newOffset := decoder.InputOffset()
+		if newOffset-currentOffset > 64*1024 {
+			_ = p.host.Log(ctx, "info", urn, fmt.Sprintf("Read %d KiB from Docker", (newOffset-currentOffset)/1024))
```

Comment:
Would we like to remove this, as it is mostly for internal diagnostics? I'm happy also to keep for now but remove later.